src/CatalogApplication/Models/README.md

```diff
@@ -0,0 +1,54 @@
+# UML Model For Tags/Ratings
+![alt text](https://i.imgur.com/eMo0eor.png "UML Diagram")
+
+# Tag Attributes
+```json
+Tag(
+    id:          int      # Generated by us somehow
+    name:        str      # The actual tag name
+    color:       int      # Hex value
+    note_id:     int      # Note ID associated with this tag
+    hidden:      bool     # Should the tag be shown
+    is_system:   bool     # Was this tag automatically added by us?
+    category:    str      # Engineering, Chemistry, CS, etc?
+)
+```
+
+## unique_id
+This is an integer generated by +1 increment
+
+## name
+A string representing the literal name of the tag i.e data-structures
+
+## color
+Hexadecimal integer value representing "good" or "bad" tag
+
+## note_id
+Integer note ID associated with the tag
```

Comment:
Be careful using an integer to save another services ID. Depending on what that team uses, it maybe an alpha-numeric meaning there are dashes and characters. You can only save that in a string.

Reply:
We have yet to consult with the notes folks to see how they are storing their data, this will likely be changing depending on their data format.